src/screens/joiningscreen.cpp

```diff
@@ -1292,7 +1309,11 @@ bool WzJoiningGameScreen_HandlerRoot::joiningSocketNETsend()
 		debug(LOG_ERROR, "Failed to send message (type: %" PRIu8 ", rawLen: %zu, compressedRawLen: %zu) to host: %s", message->type, message->rawLen(), compressedRawLen, writeErrMsg.c_str());
 		return false;
 	}
-	client_transient_socket->flush(nullptr);  // Make sure the message was completely sent.
+	// Make sure the message was completely sent.
+	if (!client_transient_socket->flush(nullptr).has_value())
+	{
+		return false;
+	}
 	ASSERT(queue->numMessagesForNet() == 0, "Queue not empty (%u messages remaining).", queue->numMessagesForNet());
```

Comment:
```suggestion
	ASSERT(queue->numMessagesForNet() == 0, "Queue not empty (%u messages remaining).", queue->numMessagesForNet());
	// Make sure the message was completely sent.
	if (!client_transient_socket->flush(nullptr).has_value())
	{
		return false;
	}
```
Let's move the `ASSERT` before the call to `flush()`